SpatialGDK/Source/SpatialGDK/Private/Interop/Startup/ElectGsmAuthWorkerStep.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "Containers/Array.h"
+#include "Misc/Optional.h"
+#include "Templates/SharedPointer.h"
+#include "Templates/UniquePtr.h"
+
+#include "Interop/Startup/SpatialStartupCommon.h"
+
+class UGlobalStateManager;
+
+namespace SpatialGDK
+{
+struct FInternalState;
+
+class FElectGsmAuthWorkerStep : public FStartupStep
+{
+public:
+	FElectGsmAuthWorkerStep(TSharedRef<FInternalState> InState, UGlobalStateManager& InGlobalStateManager);
+
+	virtual bool TryFinish() override;
+	virtual FString Describe() const override { return Executor.Describe(); }
+
+private:
+	TArray<TUniquePtr<FStartupStep>> CreateSteps();
+
+	TSharedRef<FInternalState> State;
+	UGlobalStateManager* GlobalStateManager;
+
+	TSharedRef<TOptional<bool>> bShouldHaveGsmAuthority = MakeShared<TOptional<bool>>();
```

Comment:
TSharedRef to something that is not a TSharedFromThis, dangerous.

Reply:
After applying https://github.com/spatialos/UnrealGDK/pull/3281#discussion_r683310953, this doesn't need to be a SharedRef anyway.